torch/csrc/jit/operator.cpp

```diff
@@ -260,18 +264,27 @@ struct OperatorRegistry  {
   // by performing a lookup in the operators_by_sig map.
   std::unordered_map<std::string, std::shared_ptr<Operator>> operators_by_sig;
   std::unordered_map<const char *, std::shared_ptr<Operator>> operators_by_sig_literal;
-  void registerOperator(Operator&& op){
-    std::lock_guard<std::mutex> guard(lock);
 
-    Symbol sym = Symbol::fromQualString(op.schema.name);
-    auto op_ptr = std::make_shared<Operator>(std::move(op));
-
-    operators[sym].push_back(op_ptr);
+  // XXX - caller must be holding lock
+  void registerPendingOperators() {
+    for(auto op : to_register) {
+      Symbol sym = Symbol::fromQualString(op->schema().name);
+      operators[sym].push_back(op);
+      operators_by_sig[canonicalSchemaString(op->schema())] = op;
+    }
+    to_register.clear();
+  }
 
-    operators_by_sig[canonicalSchemaString(op.schema)] = op_ptr;
+public:
+  void registerOperator(Operator&& op) {
+    std::lock_guard<std::mutex> guard(lock);
+    to_register.push_back(std::make_shared<Operator>(std::move(op)));
   }
 
+
   Operator& lookupByLiteral(const char * name) {
+    std::lock_guard<std::mutex> guard(lock);
```

Comment:
This lock was missing eventhough operators_by_sig_literal was being written to in this functions. I doubt an uncontended lock will add much overhead here.